presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -167,20 +168,34 @@ private static ConnectorPageSource createParquetPageSource(
             FileMetaData fileMetaData = parquetMetadata.getFileMetaData();
             MessageType fileSchema = fileMetaData.getSchema();
 
-            // We need to transform columns so they have the parquet column name and not table column name.
-            // In order to make that transformation we need to pass the iceberg schema (not the hive schema)
-            // in split and use that here to map from iceberg schema column name to ID, lookup parquet column
-            // with same ID and all of its children and use the index of all those columns as requested schema.
+            List<HiveColumnHandle> regularColumns = columns.stream()
+                    .filter(column -> column.getColumnType() == REGULAR)
+                    .collect(toImmutableList());
 
-            Map<String, HiveColumnHandle> parquetColumns = convertToParquetNames(columns, icebergNameToId, fileSchema);
+            // Mapping from Iceberg field ID to Parquet fields.
+            Map<Integer, org.apache.parquet.schema.Type> parquetIdToField = fileSchema.getFields().stream()
+                    .filter(field -> field.getId() != null)
+                    .collect(toImmutableMap(field -> field.getId().intValue(), Function.identity()));
 
-            List<Type> fields = parquetColumns.values().stream()
-                    .filter(column -> column.getColumnType() == REGULAR)
-                    .map(column -> getParquetType(column, fileSchema, true)) // we always use parquet column names in case of iceberg.
-                    .filter(Objects::nonNull)
+            List<org.apache.parquet.schema.Type> parquetFields = regularColumns.stream()
+                    .map(column -> {
+                        String columnName = column.getName();
+                        Integer id = icebergNameToId.get(columnName);
+                        checkState(id != null, "invalid column name: %s", columnName);
```

Comment:
Use `checkArgument` since this is an invalid argument. Let's also use a more specific error
```java
checkArgument(id != null, "column name not in ID map: %s", columnName);